File: subly_ios/Subly/Subly/Domain/PasswordValidator.swift
import Foundation

/// Minimum bar for a "strong" password: 8+ characters, at least one
/// uppercase letter, one lowercase letter, and one digit. Mirrors Android's
/// `PasswordValidator` — keep the two in sync if the rule changes.
enum PasswordValidator {
    static let minLength = 8

    static func errors(_ password: String) -> [String] {
        var errors: [String] = []
        if password.count < minLength { errors.append("At least \(minLength) characters") }
        if !password.contains(where: { $0.isUppercase }) { errors.append("At least one uppercase letter") }
        if !password.contains(where: { $0.isLowercase }) { errors.append("At least one lowercase letter") }
        if !password.contains(where: { $0.isNumber }) { errors.append("At least one number") }
        return errors
    }

    static func isStrong(_ password: String) -> Bool { errors(password).isEmpty }
}

File: subly_android/app/src/main/java/net/cynreub/subly/ui/auth/PasswordValidator.kt
package net.cynreub.subly.ui.auth

/**
 * Minimum bar for a "strong" password: 8+ characters, at least one
 * uppercase letter, one lowercase letter, and one digit.
 */
object PasswordValidator {
    private const val MIN_LENGTH = 8

    fun errors(password: String): List<String> {
        val errors = mutableListOf<String>()
        if (password.length < MIN_LENGTH) errors += "At least $MIN_LENGTH characters"
        if (password.none { it.isUpperCase() }) errors += "At least one uppercase letter"
        if (password.none { it.isLowerCase() }) errors += "At least one lowercase letter"
        if (password.none { it.isDigit() }) errors += "At least one number"
        return errors
    }

    fun isStrong(password: String): Boolean = errors(password).isEmpty()
}
